src/olympia/addons/models.py

```diff
@@ -1387,8 +1387,8 @@ def can_review(self, user):
 
     @property
     def all_dependencies(self):
-        """Return all the add-ons this add-on depends on."""
-        return list(self.dependencies.all()[:3])
+        """Return all the (valid) add-ons this add-on depends on."""
```

Comment:
I don't think we need the parens here since it's a hard fact that we only return valid add-ons.